# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoStreamFactory.cpp
# Commit: 3d0e6a5fdfb9
# Full Hash: 3d0e6a5fdfb9f2a0618e4dcd8637b2369259f09e
# Author: serge-sans-paille <sguelton@mozilla.com>
# Date: 2024-11-06 16:51:31
# Regressor Bug: 1929213
# File Overlap Count: 1
# Description:
#   Bug 1929213 - Use std::clamp instead of min/max chain in dom/media/webrtc r=bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227936
# ==============================================================================

diff -r 66fa16b4e99e -r 3d0e6a5fdfb9 dom/media/webrtc/libwebrtcglue/VideoStreamFactory.cpp
--- a/dom/media/webrtc/libwebrtcglue/VideoStreamFactory.cpp	Wed Nov 06 10:14:24 2024 +0000
+++ b/dom/media/webrtc/libwebrtcglue/VideoStreamFactory.cpp	Wed Nov 06 10:15:25 2024 +0000
@@ -136,7 +136,7 @@
   if (out_min > out_max) {
     out_min = out_max;
   }
-  out_start = std::min(out_max, std::max(out_start, out_min));
+  out_start = std::clamp(out_start, out_min, out_max);
 
   MOZ_ASSERT(pref_cap == 0 || out_max <= pref_cap);
 }
@@ -196,8 +196,8 @@
 }
 
 std::vector<webrtc::VideoStream> VideoStreamFactory::CreateEncoderStreams(
-    const webrtc::FieldTrialsView& field_trials,
-    int aWidth, int aHeight, const webrtc::VideoEncoderConfig& aConfig) {
+    const webrtc::FieldTrialsView& field_trials, int aWidth, int aHeight,
+    const webrtc::VideoEncoderConfig& aConfig) {
   // We only allow one layer when screensharing
   const size_t streamCount =
       mCodecMode == webrtc::VideoCodecMode::kScreensharing
